aptos-move/aptos-gas/src/algebra.rs

```diff
@@ -0,0 +1,226 @@
+use serde::{Deserialize, Serialize};
+use std::cmp::{Eq, Ord, Ordering, PartialEq, PartialOrd};
+use std::convert::From;
+use std::fmt::{self, Debug, Display};
+use std::marker::PhantomData;
+use std::ops::{Add, AddAssign, Mul, Sub, SubAssign};
+
+/***************************************************************************************************
+ * Units of Measurement
+ *
+ **************************************************************************************************/
+/// Unit of (external) gas.
+pub enum GasUnit {}
+
+/// Unit of internal gas.
+pub enum InternalGasUnit {}
+
+/// Unit for counting bytes.
+pub enum Byte {}
+
+/// Unit of gas currency. 1 octa = 10^-8 Aptos coins.
+pub enum Octa {}
+
+// Unit of abstract memory usage in the Move VM.
+pub enum AbstractMemoryUnit {}
+
+/// A drived unit resulted from the division of two given units.
```

Comment:
derived. Maybe easier to understand if the give the example as `UnitDiv<GasUnit, Byte>`

Reply:
I should seriously consider installing a spellchecker plugin.